packages/block-library/src/block/test/edit.native.js

```diff
@@ -0,0 +1,147 @@
+	within,
+} from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import fetchRequest from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { registerCoreBlocks } from '../..';
+
+const getMockedReusableBlock = ( id ) => ( {
+	content: {
+		raw: `
+    <!-- wp:heading -->
+    <h2>First Reusable block</h2>
+    <!-- /wp:heading -->
+    
+    <!-- wp:paragraph -->
+    <p><strong>Bold</strong> <em>Italic</em> <s>Striked</s> Superscript<sup>(1)</sup> Subscript<sub>(2)</sub> <a href="http://www.wordpress.org" target="_blank" rel="noreferrer noopener">Link</a></p>
+    <!-- /wp:paragraph -->
+    
+    !-- wp:heading {"level":4} -->
+    <h4>List</h4>
+    <!-- /wp:heading -->
+    
+    <!-- wp:list -->
+    <ul><li>First Item</li><li>Second Item</li><li>Third Item</li></ul>
+    <!-- /wp:list -->
+`,
+	},
+	id,
+	title: { raw: `Reusable block - ${ id }` },
+	type: 'wp_block',
+} );
+
+beforeAll( () => {
+	// Register all core blocks
+	registerCoreBlocks();
+} );
+
+afterAll( () => {
+	// Clean up registered blocks
+	getBlockTypes().forEach( ( block ) => {
+		unregisterBlockType( block.name );
+	} );
+} );
+
+describe( 'Reusable block', () => {
+	it( 'renders warning when the block does not exist', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const id = 1;
+		const initialHtml = `<!-- wp:block {"ref":${ id }} /-->`;
+		const endpoint = `/wp/v2/blocks/${ id }`;
+
+		const { getByA11yLabel } = await initializeEditor( {
+			initialHtml,
+		} );
+
+		const reusableBlock = await waitFor( () =>
+			getByA11yLabel( /Reusable block Block\. Row 1/ )
+		);
+
+		// Wait for the block to be fetched.
+		const reusableBlockRequest = await waitFor( () =>
+			fetchRequest.mock.calls.find( ( call ) =>
+				call[ 0 ].path.startsWith( endpoint )
+			)
+		);
+
+		const blockDeleted = await waitFor( () =>
+			within( reusableBlock ).getByText(
+				'Block has been deleted or is unavailable.'
+			)
+		);
+
+		expect( reusableBlockRequest ).toBeDefined();
+		expect( reusableBlock ).toBeDefined();
+		expect( blockDeleted ).toBeDefined();
+		expect( getEditorHtml() ).toBe( initialHtml );
+	} );
+
+	it( 'renders block content', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const id = 2;
+		const initialHtml = `<!-- wp:block {"ref":${ id }} /-->`;
+		const endpoint = `/wp/v2/blocks/${ id }`;
+
+		// Return mocked response for the block endpoint.
+		fetchRequest.mockImplementation( ( { path } ) => {
+			if ( path.startsWith( endpoint ) ) {
+				return Promise.resolve( getMockedReusableBlock( id ) );
+			}
+		} );
+
+		const { getByA11yLabel } = await initializeEditor( {
+			initialHtml,
+		} );
+
+		// Wait for the block to be fetched.
+		const reusableBlockRequest = await waitFor( () =>
+			fetchRequest.mock.calls.find( ( call ) =>
+				call[ 0 ].path.startsWith( endpoint )
+			)
+		);
```

Comment:
Do we need use `waitFor` or could this be simplified to the following? What asynchronous work is occurring here? 
```suggestion
		const reusableBlockRequest = fetchRequest.mock.calls.find( ( call ) =>
			call[ 0 ].path.startsWith( endpoint )
		);
```

Reply:
I applied this suggestion in this [commit](https://github.com/WordPress/gutenberg/pull/32038/commits/bbc4994e00f066de3852fa22abf3deb8c99d1a40).